demo/admin/src/formBuilder/blocks/fields/CheckboxListBlock.tsx

```diff
@@ -0,0 +1,140 @@
+
+const ItemBlock: BlockInterface = {
+    ...createBlockSkeleton(),
+    name: "CheckboxItem",
+    displayName: <FormattedMessage id="formBuilder.checkboxItemBlock.displayName" defaultMessage="Checkbox Item" />,
+    previewContent: (state) => [{ type: "text", content: state.label.editorState.getCurrentContent().getPlainText() }],
+    isValid: (state) => Boolean(state.fieldName),
+    input2State: (input) => ({
+        ...input,
+        label: RichTextBlock.input2State(input.label),
+        helperText: RichTextBlock.input2State(input.helperText),
+    }),
+    state2Output: (state) => ({
+        ...state,
+        label: RichTextBlock.state2Output(state.label),
+        helperText: RichTextBlock.state2Output(state.helperText),
+    }),
+    output2State: async (output, context) => ({
+        ...output,
+        label: await RichTextBlock.output2State(output.label, context),
+        helperText: await RichTextBlock.output2State(output.helperText, context),
+    }),
+    createPreviewState: (state, previewCtx) => ({
+        ...state,
+        label: RichTextBlock.createPreviewState(state.label, previewCtx),
+        helperText: RichTextBlock.createPreviewState(state.helperText, previewCtx),
+    }),
+    defaultValues: () => ({
+        label: RichTextBlock.defaultValues(),
+        helperText: RichTextBlock.defaultValues(),
+        fieldName: "",
+        mandatory: false,
+    }),
+    AdminComponent: ({ state, updateState }) => {
+        return (
+            <BlocksFinalForm onSubmit={updateState} initialValues={state}>
+                <DisplayFieldGroup>
+                    <RichTextBlockField name="label" label={<FormattedMessage id="formBuilder.checkboxItemBlock.label" defaultMessage="Label" />} />
+                    <HelperTextBlockField />
+                </DisplayFieldGroup>
+                <PropsAndValidationGroupFields />
+            </BlocksFinalForm>
+        );
+    },
+};
+
+const ItemsBlock: BlockInterface = createListBlock({
+    name: "CheckboxItems",
+    displayName: <FormattedMessage id="formBuilder.checkboxItemsBlock.displayName" defaultMessage="Checkbox Items" />,
+    block: ItemBlock,
+    itemName: <FormattedMessage id="formBuilder.checkboxItemsBlock.itemName" defaultMessage="item" />,
+    itemsName: <FormattedMessage id="formBuilder.checkboxItemsBlock.itemsName" defaultMessage="items" />,
+});
+
+const OriginalAdminComponent = ItemsBlock.AdminComponent;
+const ItemsBlockAdminComponent: BlockAdminComponent<CheckboxItemsBlockData> = (props) => {
+    const fieldNames = props.state.blocks
+        .filter(({ visible, props }) => visible && props.fieldName)
+        .map(({ props }) => props.fieldName)
+        .filter((fieldName) => fieldName !== undefined);
+
+    return (
+        <FieldNamesContext.Provider value={fieldNames}>
+            <OriginalAdminComponent {...props} />
+        </FieldNamesContext.Provider>
+    );
+};
+ItemsBlock.AdminComponent = ItemsBlockAdminComponent;
```

Comment:
This is hacky and probably shouldn't be done in a final version.

Reply:
True, I added an item to the TODO list I will copy to the feature-branch PR in https://github.com/vivid-planet/comet/pull/3040

> Find a better solution for FieldNamesContext inside AdminComponents: https://github.com/vivid-planet/comet/pull/3055#discussion_r1918797982